src/lib/BindToStores.react.js

```diff
@@ -1,47 +1,121 @@
+  function capitalize(word, pascal = true) {
+    return (pascal ? word.charAt(0).toUpperCase() : word.charAt(0)) + word.slice(1).replace(/(\_\w)/g, match => match[1].toUpperCase())
+  }
+  function storeNames() { return Object.getOwnPropertyNames(stores) }
 
-  var actionType = actionClass.Types[`GOT_${resourceName.toUpperCase()}`]
-  var resourceStore = StorePrototype(actionType)
-  var stores = {}
-  stores[resourceName] = resourceStore
+  var resourceConfigs = (Array.isArray(resources) ? resources : [resources])  // Handle old, one-resource pattern too.
+
+  var bootObjects = []  // Objects containing all necessary information for firing the bootAction of each resource
+    // and the tracking of the loading state of each resource.
+    // Example of the bootObject for an 'item' resource:
+       // {bootAction: 'fetchClient', actionClass: 'ClientActions', resourceType: 'item', resourceId: '12'}
+
+  // Construct a bootObject for every resource:
+  resourceConfigs.forEach(resourceConfig => {
+    const resourceType = Object.getOwnPropertyNames(resourceConfig)[0]
+    const resourceName = resourceConfig[resourceType]
+
+    // Get the action class--previously registered with the Container--and action Type used to fetch the resource data:
+    const actionClass = Container.getAction(resourceName)
+    const actionType = actionClass.Types[`GOT_${resourceName.toUpperCase()}`]
+
+    // Get or create the store used to hold the resource data:
+    var resourceStore = Container.getStore(resourceName)
+    if (!resourceStore)
+      resourceStore = Container.registerStore(resourceName, storePrototype(actionType))
+
+    // Add the store to the list of store the BoundComponent will listen to:
+    stores[resourceName] = resourceStore
+
+    bootObjects.push({
+      actionClass,
+      bootAction: `fetch${capitalize(resourceName)}`,
+      loadingName: `loading${capitalize(resourceName)}`,
+      resourceId: (resourceId ? resourceId : `${capitalize(resourceName, false)}Id`),
+      resourceType
+    })
+  })
 
   const BoundComponent = React.createClass({
+    contextTypes: {router: React.PropTypes.object},
     getInitialState() {
-      return {loading: true, processing: false}
+      return {
+        loading: true,
+        processing: false
+      }
     },
     componentWillMount() {
-      extractStores().forEach(store =>
-        store.addChangeListener(this.handleStoresChanged)
-      )
-      actionClass[bootAction]()
+      // Create a listener for each resource store:
+      storeNames().forEach(store => {
+        const onChangeHandlerName = `handle${store}Change`
+        this[onChangeHandlerName] = this.onChangeFactory(store)
+        stores[store].addChangeListener(this[onChangeHandlerName])
+      })
+
+      // Fire the bootAction to fetch each resource:
+      var loadingStates = {}
+      bootObjects.forEach(bootObject => {
+        loadingStates[bootObject.loadingName] = true
+
+        /* eslint-disable indent */
```

Comment:
Still want to figure out how to get ESLint to ignore the stupid way some people think `switch` indentation should be done, but figured this was not the time.